src/typemap/src/type_conv.rs

```diff
@@ -37,3 +40,67 @@ pub fn get_pipearray<'a>(arg: u64, arg_cageid: u64, cageid: u64) -> Result<&'a m
         "input data not valid",
     ));
 }
+
+/// converts a provided sockaddr pointer to a local `libc::sockaddr` pointer,
+/// and returns the pointer and its length(for ipv4, ipv6). If the socket is UNIX, the path
+/// is modified to include a LIND_ROOT prefix.
+pub fn get_sockaddr(addr: *mut u8) -> (*mut libc::sockaddr, u32) {
```

Comment:
seeing as get_sockaddr is already the name of a system call I think we should name this something else?